taichi/runtime/cuda/aot_module_builder_impl.cpp

```diff
@@ -10,8 +10,8 @@ namespace lang {
 namespace cuda {
 
 LLVMCompiledData AotModuleBuilderImpl::compile_kernel(Kernel *kernel) {
-  auto cgen = KernelCodeGenCUDA::make_codegen_llvm(kernel, /*ir=*/nullptr);
-  return cgen->run_compilation();
+  auto cgen = KernelCodeGenCUDA(kernel);
+  return std::move(cgen.compile_kernel_to_module()[0]);
```

Comment:
OOC: why do we only return std::move the first one? (do we happen to assume vector.size() == 1? if so we should assert it and probably add a note/comment

Reply:
We assume the size=1, and I will refactor it very soon in the future PRs.